ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/jdbc/JDBCImple_driver.java

```diff
@@ -614,6 +658,13 @@ public boolean write_state(Uid objUid, String typeName,
 						// Ignore
 					}
 				}
+	            if (connection != null) {
+	                try {
+	                    connection.close();
+	                } catch (SQLException e) {
+	                    // Ignore
+	                }
+	            }
 			}
 
 		}
```

Comment:
The initialise method on line 681 of your change set contains some code that creates a connection and statement. If the statement close throws an exception then the connection will be left open.